homeassistant/components/homewizard/button.py

```diff
@@ -0,0 +1,53 @@
+"""Support for HomeWizard buttons."""
+
+import logging
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Identify button."""
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    features = await coordinator.api.features()
+    if features.has_identify:
+        async_add_entities([HomeWizardIdentifyButton(coordinator, entry)])
+
+
+class HomeWizardIdentifyButton(
+    CoordinatorEntity[HWEnergyDeviceUpdateCoordinator], ButtonEntity
+):
+    """Representation of a identify button."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize button."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{entry.unique_id}_identify"
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, coordinator.data["device"].serial)},
```

Comment:
This fixes the failing tests. When not added the button is not linked to the device.

I have no idea why it does not work, or why it _does_ works here https://github.com/DCSBL/core/blob/d7f0b904d078e132146723df4a1bd16bccba8de0/homeassistant/components/homewizard/number.py#L51-L53

```suggestion
        self._attr_device_info = {
            "name": entry.title,
            "manufacturer": "HomeWizard",
            "sw_version": coordinator.data["device"].firmware_version,
            "model": coordinator.data["device"].product_type,
            "identifiers": {(DOMAIN, coordinator.data["device"].serial)},
```

Reply:
So if you don’t add the full info, the full info still needs to be given to HA via another entity. You can either have it in each entity, or you create the device as part of setting up the config entry and let each entity link to it.

Reply:
Do you accept this change now and that I open a follow-up PR with fixes for all existing HomeWizard platforms?

Reply:
yeah